### PR TITLE
Add support for loading multiplexers in KCD format

### PR DESCRIPTION
**Pull request by josko7452**
I needed a support for multiplexing for KCD file format, this PR adds just that.

### CODE REVIEW

**Comment by eerimoq** [comment]
Have a look at the failing tests and I'll review once they pass.

File: cantools/db/formats/kcd.py [hunk 1]
```diff
@@ -124,6 +124,20 @@ def _load_signal_element(signal):
                   is_float=is_float)
 
 
+def _load_mux_element(mux):
+    mux_signal = _load_signal_element(mux)
+    mux_signal._is_multiplexer = True
```
Comment (eerimoq) [inline cantools/db/formats/kcd.py]: Don't set private members.  Do `mux_siganl.is_multiplexer = ...` instead.

File: cantools/db/formats/kcd.py [hunk 2]
```diff
@@ -124,6 +124,20 @@ def _load_signal_element(signal):
...
+    signals = [mux_signal]
+    for muxgrp in mux.findall('ns:MuxGroup', NAMESPACES):
+        ids = muxgrp.attrib['count']
+        for signal in muxgrp.findall('ns:Signal', NAMESPACES):
+            m = _load_signal_element(signal)
```
Comment (eerimoq) [inline cantools/db/formats/kcd.py]: Variable name `m` too short. Make it readable and searchable.

File: cantools/db/formats/kcd.py [hunk 3]
```diff
@@ -124,6 +124,20 @@ def _load_signal_element(signal):
                   is_float=is_float)
 
 
+def _load_mux_element(mux):
+    mux_signal = _load_signal_element(mux)
+    mux_signal._is_multiplexer = True
+    signals = [mux_signal]
+    for muxgrp in mux.findall('ns:MuxGroup', NAMESPACES):
+        ids = muxgrp.attrib['count']
```
Comment (eerimoq) [inline cantools/db/formats/kcd.py]: Should be singular, not plural form? Rename to `multiplexer_id`?

File: cantools/db/formats/kcd.py [hunk 4]
```diff
@@ -168,8 +182,12 @@ def _load_message_element(message, bus_name, nodes):
             senders.append(_get_node_name_by_id(nodes,
                                                 sender.attrib['id']))
 
-    # Find all signals in this message.
     signals = []
+    # Multiplexers
```
Comment (eerimoq) [inline cantools/db/formats/kcd.py]: Add a blank line before the comment.

File: cantools/db/formats/kcd.py [hunk 5]
```diff
@@ -168,8 +182,12 @@ def _load_message_element(message, bus_name, nodes):
...
+    for mux in message.findall('ns:Multiplex', NAMESPACES):
+        signals += _load_mux_element(mux)
+
+    # Find all signals in this message.
```
Comment (eerimoq) [inline cantools/db/formats/kcd.py]: Remove the blank line after the comment.

**Comment by eerimoq** [comment]
@josko7452 Please have a look at my comments.

**Comment by josko7452** [comment]
@eerimoq I was bit busy with other stuff.. Thanks for the comments I plan to rectify it either on Sunday or Monday.. Cheers.

**Comment by eerimoq** [comment]
Ok, thanks!

**Comment by josko7452** [comment]
Hi @eerimoq I think I have addressed all the comments. Cheers.

File: tests/test_database.py [hunk 6]
```diff
@@ -767,7 +767,26 @@ def test_the_homer(self):
...
+        self.assertEqual(info_0.is_float, False)
+
+        abs_info_mux = db.messages[1].signals[8]
+        self.assertEqual(abs_info_mux.name, 'ABS_InfoMux')
+        self.assertEqual(abs_info_mux.start, 16)
+        self.assertEqual(abs_info_mux.length, 2)
+        self.assertEqual(abs_info_mux.receivers, [])
+        self.assertEqual(abs_info_mux.byte_order, 'little_endian')
+        self.assertEqual(abs_info_mux.is_signed, False)
+        self.assertEqual(abs_info_mux.is_float, False)
```
Comment (eerimoq) [inline tests/test_database.py]: Verify the multiplexer attributes as well, as those are an important part of the new functionality.

Comment (josko7452) [reply tests/test_database.py]: Good point. I have added it there.

**Comment by eerimoq** [comment]
Looks good, thanks!

**Comment by eerimoq** [comment]
Part of release 23.1.0, uploaded to PyPI!

**Pull request merged**